phpBB/includes/functions_admin.php

```diff
@@ -2573,6 +2573,11 @@ function prune($forum_id, $prune_mode, $prune_date, $prune_flags = 0, $auto_sync
 		$sql_and .= ' AND topic_status = ' . ITEM_MOVED . " AND topic_last_post_time < $prune_date";
 	}
 
+	if ($prune_limit > 0)
+	{
+		$sql_and .= " LIMIT $prune_limit";
```

Comment:
Updating the event should be fairly straightforward. Just add another line with
`@var int prune_limit The prune limit` and also add a `@changed 3.1.10-RC1 Added prune_limit` after the line with `@since`. Then you'll also have to add `'prune_limit'` to the `$vars` array and probably change it to be in multi-line notation due to the amount of variables like this:
```
$vars = array(
    'forum_id',
    'prune_mode',
    'prune_date',
    'prune_flags',
    'auto_sync',
    'sql_and',
    'prune_limit'
);
```
And that should be it.